wsd/DocumentBroker.cpp

```diff
@@ -2118,10 +2118,35 @@ DocumentBroker::NeedToSave DocumentBroker::needToSaveToDisk() const
     return NeedToSave::No;
 }
 
+bool DocumentBroker::manualSave(const std::shared_ptr<ClientSession>& session,
+                                bool dontTerminateEdit, bool dontSaveIfUnmodified,
+                                const std::string& extendedData)
+{
+    // If we aren't saving already.
+    if (_docState.activity() != DocumentState::Activity::Save)
```

Comment:
silly question - how does this 'manual' save get re-dispatched later when we are not saving ? =)

Reply:
If the user wants to save again in the future, they can use any of the shortcuts to do so. This only prevents multiple saves in flight. (Unless I misunderstood the question.)

Reply:
Hmm - but if I have a large document, hit save, while it is saving (or at least up-loading) - I type "Sell everything right now!" and hit save - the save will effectively not happen; is that so ? - hopefully it won't loose my last modification in the long run, but ... is that expected? and/or will we get the right modification status in the UI - ultimately, as long as the UI says it is not saved - I suppose that's ok for now.

Reply:
Correct. The document will still say modified. I see your concern, though, in that we don't want to lose the last modification. That shouldn't happen even before this change, as we are more aggressive with saves, even when modified=false, if we had input from the user, we still save. This PR makes the modified flag more reliable and the reduces in-flight save commands to avoid unnecessary duplicates.

Notice that uploading doesn't affect saving, so that could still happen in parallel.

Reply:
Ah - it is fine; it is a "de-bounce" feature; and caveat user etc. =) just wanted to understand; lets not over-complicate things by handling it.